app/models/ems_refresh/save_inventory_infra.rb

```diff
@@ -44,6 +44,19 @@ def save_ems_infra_inventory(ems, hashes, target = nil)
       return
     end
 
+    # Check if a vm was removed and we still have disk(s)
+    if target.kind_of?(VmOrTemplate) && hashes[:vms].blank?
+      current = hashes[:vm_disks].collect { |disk| URI(disk[:storage][:ems_ref]).path.split('/').last }.uniq
+      if current.blank?
+        storage = nil
+      else
+        vm_storages = ([target.storage] + target.storages).compact.uniq
+        storage = vm_storages.select { |store| !current.include?(store.ems_ref) }
+      end
+
+      target.disconnect_storage storage
+    end
```

Comment:
I am open to do it in different place. Before this change we run this very method using automate [user_remove_vm_finished.yaml](https://github.com/ManageIQ/manageiq/blob/master/db/fixtures/ae_datastore/ManageIQ/System/Event/EmsEvent/RHEVM.class/user_remove_vm_finished.yaml#L12). This method never checks whether the disks are there after the removal or not. As a result we always archive a vm. It looks like what I am missing here is making sure it is triggered only for RHV.